databricks/sdk/mixins/sql.py

```diff
@@ -0,0 +1,326 @@
+
+        .. code-block::
+
+            iterate_rows = functools.partial(w.statement_execution.iterate_rows, warehouse_id, catalog='samples')
+            for row in iterate_rows('SELECT * FROM nyctaxi.trips LIMIT 10'):
+                pickup_time, dropoff_time = row[0], row[1]
+                pickup_zip = row.pickup_zip
+                dropoff_zip = row['dropoff_zip']
+                all_fields = row.as_dict()
+                print(f'{pickup_zip}@{pickup_time} -> {dropoff_zip}@{dropoff_time}: {all_fields}')
+
+        :param warehouse_id: str
+          Warehouse upon which to execute a statement.
+        :param statement: str
+          SQL statement to execute
+        :param byte_limit: int (optional)
+          Applies the given byte limit to the statement's result size. Byte counts are based on internal
+          representations and may not match measurable sizes in the JSON format.
+        :param catalog: str (optional)
+          Sets default catalog for statement execution, similar to `USE CATALOG` in SQL.
+        :param schema: str (optional)
+          Sets default schema for statement execution, similar to `USE SCHEMA` in SQL.
+        :param timeout: timedelta (optional)
+          Timeout after which the query is cancelled. If timeout is less than 50 seconds,
+          it is handled on the server side. If the timeout is greater than 50 seconds,
+          Databricks SDK for Python cancels the statement execution and throws `TimeoutError`.
+        :return: Iterator[Row]
+        """
+        execute_response = self.execute(warehouse_id,
+                                        statement,
+                                        byte_limit=byte_limit,
+                                        catalog=catalog,
+                                        schema=schema,
+                                        timeout=timeout)
+        if execute_response.result.external_links is None:
+            return []
+        for row in self._iterate_external_disposition(execute_response):
+            yield row
+
+    def _result_schema(self, execute_response: ExecuteStatementResponse):
+        col_names = []
+        col_conv = []
+        for col in execute_response.manifest.schema.columns:
+            col_names.append(col.name)
+            conv = self._type_converters.get(col.type_name, None)
+            if conv is None:
+                msg = f"{col.name} has no {col.type_name.value} converter"
+                raise ValueError(msg)
+            col_conv.append(conv)
+        row_factory = functools.partial(Row, col_names)
+        return row_factory, col_conv
+
+    def _iterate_external_disposition(self, execute_response: ExecuteStatementResponse) -> Iterator[Row]:
+        # ensure that we close the HTTP session after fetching the external links
+        result_data = execute_response.result
+        row_factory, col_conv = self._result_schema(execute_response)
+        with self._api._new_session() as http:
```

Comment:
Does this make sure we don't use the authorization header that's used for the internal requests? That is important.

Reply:
@sander-goos `self._api._new_session()` starts HTTP session without our auth headers, because pre-signed URLs don't expect any Authorization headers.

all other requests follow the normal flow.